src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
+
+type Color = 'red' | 'green' | 'blue';
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
 
+  getArea(): number
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
```

Comment:
you cheked if side <= 0 3 times in your code(27, 55 & 73 lines). create and use the function for this check!=)